### PR TITLE
Adding aggregation step to generate profile module

### PR DESCRIPTION
**Pull request by gwaygenomics**
This script is the next downstream step after #8 and uses `pycytominer.aggregate()`.

Currently, the output of this script (using CP074A as an example) are two files:

1. Gene aggregation (592KB)
2. Guide aggregation (2.4MB)

### CODE REVIEW

File: 1.generate-profiles/1.aggregate.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+)
+args = parser.parse_args()
+config_file = args.config_file
+
+config = process_config_file(config_file)
+
+# Extract config arguments
+core_args = config["core"]
+batch = core_args["batch"]
+trash_files = core_args["trash_files"]
```
Comment (ErinWeisbart) [inline 1.generate-profiles/1.aggregate.py:28]: Per our recent conversation about the site_processing_config, trash_files should become ignore_files

File: 1.generate-profiles/1.aggregate.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+if aggregate_from_single_file and aggregate_from_site:
+    warnings.warn(
+        "Both sources set to true. Defaulting to aggregate `from_single_file`"
+    )
+
+# Load single cell data
+if aggregate_from_single_file:
+    single_cell_df = pd.read_csv(single_cell_file, sep=",")
+else:
+    sites = [x for x in single_cell_output_dir.iterdir() if x.name not in trash_files]
```
Comment (ErinWeisbart) [inline 1.generate-profiles/1.aggregate.py:59]: trash_files should become ignore_files

File: 1.generate-profiles/1.aggregate.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+# Input argument flow control
+if aggregate_from_single_file:
+    assert single_cell_file.exists(), "Error! The master single cell file does not exist! Did you mean to set `from_single_file` to False?"
+else:
+    assert aggregate_from_site, "You must set one aggregate source to true in the config (either `from_site_files` or `from_single_file`)"
+
+if aggregate_from_single_file and aggregate_from_site:
+    warnings.warn(
+        "Both sources set to true. Defaulting to aggregate `from_single_file`"
```
Comment (ErinWeisbart) [inline 1.generate-profiles/1.aggregate.py:52]: More clear to say:
"Both `from_site_files` and `from_single_file` are set to true in the config. Defaulting to aggregate `from_single_file`"

**Comment by ErinWeisbart** [comment]
This calls config["aggregate"] but that's not in this PR. Should it be?

**Comment by gwaygenomics** [comment]
> This calls config["aggregate"] but that's not in this PR. Should it be?

Great catch. It should be... yes, note that I did the lazy thing and added the `aggregate` config options in https://github.com/broadinstitute/pooled-cell-painting-profiling-recipe/pull/8/commits/e606fe12d925d2aa4b046e4728452cec5aaa2640#diff-04ba2a82a13772907c6597e5ae1f54f2R53-R66

The "lazy" part means that we won't have to deal with merge conflicts by two pull requests dealing with the same file changing. In general, github makes merging _nearly_ a non-trivial problem (and this ease of asynchronous development is one of the reasons github is so great!). (It can get VERY messy too - just something to be mindful of)

**Comment by gwaygenomics** [comment]
thanks for this succinct review @ErinWeisbart - ready for your re-review when you get a chance!

**Pull request merged**